### PR TITLE
Add endpoint to delete direct messages #657

### PR DESCRIPTION
**Pull request by RageZBla**
### Make sure you've reviewed the checklist. (DO NOT REMOVE)
🚨Please review the [guidelines for contributing](https://github.com/iluminar/goodwork/wiki/Contribution-Guideline) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch**. Also you should start *your branch* off *dev branch*.
- [x] Check you have included issue number in your pr (e.g #23).
- [x] Check you have added tests that prove your fix is effective or that your feature works.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] Check you added any notable changes to the `CHANGELOG.md` file.

### Description
Add endpoint to delete direct messages, see #657 

💔Thank you!

### CODE REVIEW

File: app/Core/Policies/DirectMessagePolicy.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    /**
+     * Determine whether the user can delete the direct message.
+     *
+     * @param  DirectMessage $directMessage
+     * @param  User          $user
+     * @return bool
+     */
+    public function delete(User $user, DirectMessage $directMessage)
+    {
+        // TODO: I am not sure if it's correct
```
Comment (Hasnayeen) [inline app/Core/Policies/DirectMessagePolicy.php]: just checking the user is the owner of the message is enough, so checking message sender_id against user is should be enough. `$user->id === $directMessage->sender_id`

**Comment by RageZBla** [comment]
@Hasnayeen thanks for the hand :+1:

File: app/Core/Policies/DirectMessagePolicy.php [hunk 2]
```diff
@@ -19,6 +19,6 @@ class DirectMessagePolicy
      */
     public function delete(User $user, DirectMessage $directMessage)
     {
-        return $user->id == $directMessage->sender_id;
+        return $user->id === $directMessage->sender_id;
```
Comment (RageZBla) [inline app/Core/Policies/DirectMessagePolicy.php]: @Hasnayeen sadly it seems the type is not same (int vs string)

Comment (Hasnayeen) [reply app/Core/Policies/DirectMessagePolicy.php]: @RageZBla oh ok, that's weird. revert back to original then.

**Comment by Hasnayeen** [comment]
@RageZBla excellent work :clap: if you want to get more involved with the project you can join our discord server [here](https://discord.gg/4DvTQsc)

**Pull request merged**